omnisend/includes/SDK/V1/class-contact.php

```diff
@@ -128,13 +129,23 @@ public function to_array(): array {
 			'tags'        => array_values( array_unique( $this->tags ) ),
 		);
 
+		$email_consent_status = $this->email_opt_in_source ? 'subscribed' : 'nonSubscribed';
```

Comment:
it is incorrect to calculate this status from optin source, just leave it like this

`$email_consent_status = $this->email_status`
and SDK users will set needed property to `email_status`

Reply:
and validate this status in

`public function validate(): WP_Error` 
to have one of valid values

Reply:
It is a bit tricky situation. Let me explain.
Opt-in/opt-out is not equal to consent. It is two different things.
Opt-in/opt-out defines contact status: subscribed,nonSubscribed,unsubscribed
Consent is EU gdpr stuff, that does not influence contact status mentioned above, let's treat like additional contact information which is optional and lets not mix it with status. Consent does not have status, it only can be added.

Contact may have three different status:
subscribed,nonSubscribed,unsubscribed and it is not consent.

If status is not known (or set) it is SAFE to pass "nonSubscribed" this does NOT CHANGE existing contact status.

Currently (before this PR), there is no DIRECT way to set status.
If  phone or email opt-in source is set with functions `set_phone_opt_in`, `set_email_opt_in` then contact is passed as `subscribed`, if not set as `nonSubscribed`. Currently no way to pass as `unsubscribed`. We want to change that!

Here is my suggestion.  Statuses subscribed,nonSubscribed,unsubscribed is a bit difficult and error prone as it is text variables, we do not want to allow set it directly, let's make it simplier, so following changes are needed:
1. Change default value `private $email_status = null;` -> `private $email_status = 'nonSubscribed';`
2. Change default value `private $phone_status = null;` -> `private $phone_status = 'nonSubscribed';`
3. Remove `set_email_status`
4. Remove `set_phone_status`
5. Change `set_email_opt_in` so function also change `email_status` to `subscribed` (backward compatibility)
6. Change `set_phone_opt_in` so function also change `phone_status` to `subscribed`  (backward compatibility)
7. Add function `set_email_subscriber()` which change `email_status` to `subscribed`
8. Add function `set_phone_subscriber()` which change `phone_status` to `subscribed`
9. Add function `set_email_unsubscriber()` which change `email_status` to `unsubscribed`
10. Add function `set_phone_unsubscriber()` which change `phone_status` to `unsubscribed`
11. Instead of `$email_consent_status = $this->email_opt_in_source ? 'subscribed' : 'nonSubscribed';` just use `$this->email_status` (same for phone)

These changes should be backward compatible and no `email_status` or `phone_status` validation should be needed, because it will not be possible for SDK client incorrectly set these values.

@greta-mik please double check.

Reply:
thank you for detailed explanation and suggested changes. I did as suggested Checked looks like it's working as intended.